### PR TITLE
Avoid adding global vars that are flags to completions

### PR DESCRIPTION
**Pull request by fabianofranz**
Fixes https://github.com/openshift/origin/issues/7323
Fixes https://github.com/openshift/origin/issues/6192

[This line](https://github.com/openshift/origin/pull/7482/files#diff-cc52507c1aeaa02be3653aa53ad588f4L584 ) has the side effect of adding **every** flag that is declared as global vars (outside of a flag set) in any dependency present in Godeps. One example is the `--container-hints` flag, [declared by cAdvisor](https://github.com/google/cadvisor/blob/master/container/raw/container_hints.go#L28).

### CODE REVIEW

**Comment by smarterclayton** [comment]
I think that's ok, that was an explicit choice before.

**Comment by fabianofranz** [comment]
Related:

https://github.com/kubernetes/kubernetes/commit/bcf912667b8296c43b6837d04357da5cd7f3a08d
https://github.com/spf13/cobra/issues/44
https://github.com/spf13/cobra/pull/75

**Comment by fabianofranz** [comment]
@smarterclayton this is the reason why https://github.com/openshift/origin/issues/7323 happens. Kube is not bitten by that because they only compile the `genbashcomp` for `kubectl`, which probably does not include cAdvisor and other external deps that declare flags as global vars, while do completions for `openshift`, `oadm` and `oc`.

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/cmd.go [hunk 1]
```diff
@@ -159,6 +159,7 @@ Find more information at https://github.com/kubernetes/kubernetes.`,
 	}
 
 	f.BindFlags(cmds.PersistentFlags())
+  f.BindExternalFlags(cmds.PersistentFlags())
```
Comment (smarterclayton) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/cmd.go]: Gofmt is wrong

**Comment by fabianofranz** [comment]
@smarterclayton good to go?

**Comment by smarterclayton** [comment]
Lgtm [merge]

**Comment by smarterclayton** [comment]
hung [merge]

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
We may have to excise more:

+    flags+=("--storage-driver-buffer-duration=")
+    flags+=("--storage-driver-db=")
+    flags+=("--storage-driver-host=")
+    flags+=("--storage-driver-password=")
+    flags+=("--storage-driver-secure")

Is showing up in kube subcommands

**Pull request merged**